Sources/_StringProcessing/Engine/Registers.swift

```diff
@@ -128,6 +135,9 @@ extension Processor.Registers {
   }
 
   mutating func reset(sentinel: Input.Index) {
+    guard isDirty else {
+      return
+    }
```

Comment:
Need to reset `isDirty` here, as well.

Reply:
Shoot, let's see if that affects numbers...

Reply:
Oh, you mean we can set `isDirty = false` outside of the guard statement? Right, that will actually be an improvement.